lib/src/util/stack_trace_mapper.dart

```diff
@@ -17,14 +18,69 @@ class StackTraceMapper {
   /// The URI of the SDK root from which dart2js loaded its sources.
   final Uri _sdkRoot;
 
-  StackTraceMapper(String contents,
+  /// The contents of the source map.
+  final String _contents;
+
+  /// The URI of the source map.
+  final Uri _mapUrl;
+
+  StackTraceMapper(this._contents,
       {Uri mapUrl, SyncPackageResolver packageResolver, Uri sdkRoot})
-      : _mapping = parseExtended(contents, mapUrl: mapUrl),
+      : _mapping = parseExtended(_contents, mapUrl: mapUrl),
+        _mapUrl = mapUrl,
         _packageResolver = packageResolver,
         _sdkRoot = sdkRoot;
 
   /// Converts [trace] into a Dart stack trace.
   StackTrace mapStackTrace(StackTrace trace) =>
       mapper.mapStackTrace(_mapping, trace,
           packageResolver: _packageResolver, sdkRoot: _sdkRoot);
+
+  /// Returns a Map representation which is suitable for JSON serialization.
+  Map<String, dynamic> serialize() {
+    return {
+      'contents': _contents,
+      'sdkRoot': _sdkRoot?.toString(),
+      'packageConfigMap':
+          _serializablePackageConfigMap(_packageResolver.packageConfigMap),
+      'packageRoot': _packageResolver.packageRoot?.toString(),
+      'mapUrl': _mapUrl?.toString(),
+    };
+  }
+
+  /// Returns a Future which will resolve to a [StackTraceMapper] contained in
+  /// the provided serialized representation.
+  static Future<StackTraceMapper> deserialize(Map serialized) async {
```

Comment:
If you use the [`SyncPackageResolver`](https://www.dartdocs.org/documentation/package_resolver/1.0.2/package_resolver/SyncPackageResolver-class.html) constructors, you should be able to make this synchronous.